### PR TITLE
Add module.parent fallback

### LINKED ISSUE #17: Missing module.parent in Docker container
The strangest thing happened, everything works in Node 6 & 7, however whenever I bring `conf` into any of the latest Docker container (ex: `node:6`) it crashes.

https://github.com/sindresorhus/conf/blob/master/index.js#L13

**Error on require**
```bash
# node
> const Conf = require('conf')
TypeError: Path must be a string. Received null
    at assertPath (path.js:7:11)
    at Object.dirname (path.js:1324:5)
    at Object.<anonymous> (/usr/src/app/node_modules/conf/index.js:13:24)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
```

**Module parent does not exist**
```
> module.parent
undefined
> module
Module {
  id: '<repl>',
  exports: {},
  parent: undefined,
  filename: null,
  loaded: false,
  children: [],
  paths:
   [ '/usr/src/app/repl/node_modules',
     '/usr/src/app/node_modules',
     '/usr/src/node_modules',
     '/usr/node_modules',
     '/node_modules',
     '/root/.node_modules',
     '/root/.node_libraries',
     '/usr/local/lib/node' ] }
```

### PR DESCRIPTION
**Pull request by DenisCarriere**
Getting an linting error.

```
Unexpected mix of && and ||.
```
@sindresorhus Might need some help :)

---

Fixes #17

### CODE REVIEW

**Comment by SamVerschueren** [comment]
`module.parent ? module.parent.filename : '.'`

**Comment by DenisCarriere** [comment]
@SamVerschueren That wouldn't catch if `module.parent.filename` is `undefined`.

**Comment by DenisCarriere** [comment]
This would look at lot nicer once [`Null Propagation Operators`](https://github.com/tc39/proposals) is implemented in Javascript.

https://docs.google.com/presentation/d/11O_wIBBbZgE1bMVRJI8kGnmC6dWCBOwutbN9SWOK0fU/view#slide=id.p

File: index.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const obj = () => Object.create(null);
 
 // Prevent caching of this module so module.parent is always accurate
 delete require.cache[__filename];
-const parentDir = path.dirname(module.parent.filename || '.');
+const parentDir = path.dirname(module.parent && module.parent.filename || '.');
```
Comment (kevva) [inline index.js]: `(module.parent && module.parent.filename) || '.'`

**Comment by DenisCarriere** [comment]
@SamVerschueren Fixed this PR

**Comment by DenisCarriere** [comment]
🎉

**Pull request merged**